Sources/Display.php

```diff
@@ -851,26 +852,26 @@ function Display()
 		if (isset($_SESSION['page_next_start']) && $_SESSION['page_next_start'] == $start)
 		{
 			$start_char = 'M';
-			$page_id = $_SESSION['page_last_id'];
+			$page_id = $ascending ? $_SESSION['page_last_id'] : $_SESSION['page_first_id'];
 		}
 		// User moved to the previous page
 		elseif (isset($_SESSION['page_before_start']) && $_SESSION['page_before_start'] == $start)
 		{
 			$start_char = 'L';
-			$page_id = $_SESSION['page_first_id'];
+			$page_id = $ascending ? $_SESSION['page_first_id'] : $_SESSION['page_last_id'];
 		}
 		// User refreshed the current page
 		elseif (isset($_SESSION['page_current_start']) && $_SESSION['page_current_start'] == $start)
 		{
 			$start_char = 'C';
-			$page_id = $_SESSION['page_first_id'];
+			$page_id = $ascending ? $context['topicinfo']['id_fist_msg'] : $context['topicinfo']['id_last_msg'];
 		}
 	}
 	// Special case start page
 	elseif ($start == 0)
 	{
 		$start_char = 'C';
-		$page_id = $context['topicinfo']['id_first_msg'];
+		$page_id = $ascending ? $context['topicinfo']['id_first_msg'] : $_SESSION['page_last_id'];
```

Comment:
Why `$_SESSION['page_last_id']` instead of `$context['topicinfo']['id_last_msg'];`?